libs/ml/include/ml/model/model.hpp

```diff
@@ -490,6 +508,11 @@ void Model<TensorType>::TrainImplementation(DataType &loss, SizeType n_rounds)
   loss = loss_;
 }
 
+template <typename TensorType>
+bool Model<TensorType>::DataLoaderIsSet()
+{
+  return dataloader_ptr_->Size() != 0;
```

Comment:
is it possible for this to be called while dataloader_ptr_ is still a null_ptr?

Reply:
Good catch!